### DISCUSSION TITLE
Layouts causing issues with the keybindings

### DISCUSSION THREAD

**Discussion by arcolinuxz** [leftwm/leftwm#1124] [Q&A]
I am currently on these versions

![image](https://github.com/leftwm/leftwm/assets/36679406/95912c42-e1ac-481a-8086-14de0b2c399d)

When I change the config.ron file to include the list of layouts we have keybinding issues ( we work with sxhkd as well).
We can change the layout. When we ask a check for the .ron file we see this error.

![with layouts](https://github.com/leftwm/leftwm/assets/36679406/b40fca3a-1b43-4260-89a2-55352a826862)

When we remove the list of layouts all keybindings are working again as in 30 commits ago but we have no rotation in layouts then. When we check the .ron conf then it is valid.

![without layouts](https://github.com/leftwm/leftwm/assets/36679406/4408986f-471a-4c2f-858b-dca4d6345011)

I saw one of the commits has to do with layouts - https://github.com/leftwm/leftwm/commit/f5fa58fcb981b03adc5f2670ef28eedcb513284b

Previous build of Leftwm and it is still working perfectly - layouts and keybindings is

https://github.com/leftwm/leftwm/commit/c6cbca6ef24b840cf1337ba98b7d52c1138f1ca6

Current code of ArcoLinux Leftwm - config.ron file

https://github.com/arcolinux/arcolinux-leftwm/blob/master/etc/skel/.config/leftwm/config.ron

How can I help to debug this?

**Comment by arcolinuxz** [comment]
I think I discovered something. 
Removing the line layouts all together solves my issue.

keybindings work
layout swith works - using the default from your rs file

**Comment by VuiMuich** [comment]
Hi Erik,
with the current `git` we had a breaking change for layout configuration. Wrapping the layout names in `"` each should fix this properly.
Here is a snippet from my current config. Note the newly added possibility to define new layouts in the config (this feature made this breaking change necessary):
```rust
    layouts: [
        "MainAndDeck",
        "Monocle",
        "CenterMainFluid",
        "Foo",        
        // "LeftWiderRightStack",
    ],
    layout_definitions: [
    (
        name: "Foo",
        flip: None,
        rotate: North,
        reserve: ReserveAndCenter,
        columns: (
            flip: None,
            rotate: North,
            main: Some(Main(count: 1, size: 0.4, flip: None, rotate: North, split: Some(Horizontal))),
            stack: (flip: None, rotate: North, split: Some(Vertical)),
            second_stack: Some(SecondStack(flip: None, rotate: North, split: Horizontal)),
        )
    ),
    ],
```

**Comment by arcolinuxz** [comment]
I suggest you change the wiki to follow the new syntax with quotes (still without quotes there)

![working](https://github.com/leftwm/leftwm/assets/36679406/b1cc5330-ac47-4a35-94c8-67441b29c656)

I think I am going to leave the line 'layouts' out as we do not want to limit our users in the layout.

In doing so I believe that if you guys develop a new layout we will get it automatically.
Is that correct?

**Comment by VuiMuich** [comment]
This is absolutely the plan to update the wiki, I am currently not sure, but I believed we decided explicitly to wait with this wiki update for the release.

Not necessarily all added layouts would be added to the default config, but I guess we will decide, when a layout gets added.
If you want to stick with just the internal default, I would recommend to leave a comment, so actually configuring layouts is easier accessible to the user.
